### PR TITLE
[ARO-4453] Add MISE container to RP 

### PR DESCRIPTION
**Pull request by jonathan34c**
### Which issue this PR addresses:

Add Mise Container

Screenshot 2023-10-17 114209
I will enable mise authentication in the next PR

### CODE REVIEW

File: poc/pkg/templates/deployment.yaml [hunk 1]
```diff
@@ -47,6 +47,25 @@ spec:
               port: http
           resources:
             {{- toYaml .Values.resources | nindent 12 }}
+        - name: mise
+          image: aropocsvc.azurecr.io/mise-sidecar:dev
```
Comment (niontive) [inline poc/pkg/templates/deployment.yaml]: Let's not hardcode this name

File: poc/pkg/templates/mise-configMap.yaml [hunk 2]
```diff
@@ -0,0 +1,33 @@
+apiVersion: v1
+data:
+  appsettings.json: |
+    {
+      "Version": "1",
+      "AzureAd": {
+        "Instance": "https://login.microsoftonline.com",
+        "ClientId": "704c7dfd-4ed1-4732-95c0-04a44b0895a0",
```
Comment (niontive) [inline poc/pkg/templates/mise-configMap.yaml]: Let's not hardcode Client ID

File: poc/pkg/templates/mise-configMap.yaml [hunk 3]
```diff
@@ -0,0 +1,33 @@
+apiVersion: v1
+data:
+  appsettings.json: |
```
Comment (niontive) [inline poc/pkg/templates/mise-configMap.yaml:5]: Let's set the logging level to debug

File: poc/pkg/templates/mise-configMap.yaml [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+    {
+      "Version": "1",
+      "AzureAd": {
+        "Instance": "https://login.microsoftonline.com",
+        "ClientId": "704c7dfd-4ed1-4732-95c0-04a44b0895a0",
+        "TenantId": "72f988bf-86f1-41af-91ab-2d7cd011db47",
+        "InboundPolicies": 
+        [
+          {
+            "Label": "jonachang-policy",
```
Comment (niontive) [inline poc/pkg/templates/mise-configMap.yaml]: Let's rename this to `aro-rp-policy`, and don't hardcode

File: poc/pkg/templates/mise-configMap.yaml [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+            "AuthenticationSchemes": [ "Bearer"],
+            "ValidAudiences" : [ 
+                "https://management.azure.com/"
```
Comment (weinong) [inline poc/pkg/templates/mise-configMap.yaml]: we shouldn't use this

Comment (jonathan34c) [reply poc/pkg/templates/mise-configMap.yaml:16]: which audience should i use? just the clientid for poc purpose?

File: poc/pkg/values.yaml [hunk 6]
```diff
@@ -90,3 +90,46 @@ AKSCluster:
...
+      name: config
+      mountPath: /etc/mise
+      configMap:
+        name: miseconfig
+        items:
+          key: appsettings.json
+          path: appsettings.json
+  configMap:
+    Instance: https://login.microsoftonline.com
+    ClientId: 704c7dfd-4ed1-4732-95c0-04a44b0895a0
```
Comment (niontive) [inline poc/pkg/values.yaml]: Is this the client ID of our managed identity?

If so, let's make sure to update the release such that we dynamically set this value. For example, if we re-deploy the managed identity, this value changes.

File: poc/pkg/values.yaml [hunk 7]
```diff
@@ -90,3 +90,46 @@ AKSCluster:
...
+    TenantId: 72f988bf-86f1-41af-91ab-2d7cd011db47
+    Audience: https://management.azure.com
+    # Possible log levels: Trace, Debug, Information, Warning, Error, Critical, None
+    LogLevel: Debug
+    InboundPolicies:
+      Label: aro-rp-policy
+      AuthenticationSchemes: Bearer
+      TokenTypes: AppToken
+      Authority: https://login.microsoftonline.com/72f988bf-86f1-41af-91ab-2d7cd011db47/v2.0
+      ValidApplicationIds: 704c7dfd-4ed1-4732-95c0-04a44b0895a0
```
Comment (niontive) [inline poc/pkg/values.yaml]: Same here

File: poc/pkg/values.yaml [hunk 8]
```diff
@@ -90,3 +90,46 @@ AKSCluster:
...
+    AllowedHosts: "*"
+    Kestrel:
+      Endpoints:
+        Url: http://0.0.0.0:5000
+    metatata:
+      name: miseconfig
+    Logging:
+      LogLevel:
```
Comment (niontive) [inline poc/pkg/values.yaml:135]: Let's set these to debug as well for now

File: poc/pkg/templates/mise-configMap.yaml [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+  # https://identitydivision.visualstudio.com/DevEx/_git/Microsoft-IdentityModel-S2S?path=/docs/JsonConfigurations.md&_a=preview
+  appsettings.json: |
+    {
+      "AzureAd": {
+        "Instance": "{{ .Values.MISE.configMap.Instance }}",
+        "ClientId": "{{ .Values.MISE.configMap.ClientId }}",
+        "TenantId": "{{ .Values.MISE.configMap.TenantId }}",
+        "Audience": "{{ .Values.MISE.configMap.Audience }}",
+        "Logging": {
+          "LogLevel": "{{ .Values.MISE.configMap.LogLevel }}"
```
Comment (niontive) [inline poc/pkg/templates/mise-configMap.yaml]: Let's rename this to `.Values.MISE.configMap.AzureAD.LogLevel`

**Review by niontive — APPROVED** [review]

**Pull request merged**